File: shared/domain/configuration/services/ConfigDomainService.kt
package shared.domain.configuration.services

import shared.domain.configuration.entities.AppConfig
import shared.domain.configuration.ports.inbound.ConfigService
import shared.domain.configuration.ports.outbound.ConfigRepository

/**
 * 配置领域服务
 * Configuration 上下文的核心服务实现
 */
class ConfigDomainService(
    private val configRepository: ConfigRepository
) : ConfigService {

    override fun getTargetLanguage(): String = configRepository.getTargetLanguage()

    override fun setTargetLanguage(lang: String) {
        configRepository.setTargetLanguage(lang)
    }

    override fun getLlmProvider(): String = configRepository.getLlmProvider()

    override fun setLlmProvider(provider: String) {
        configRepository.setLlmProvider(provider)
    }

    override fun getLlmApiKey(provider: String): String? = configRepository.getLlmApiKey(provider)

    override fun setLlmApiKey(provider: String, key: String) {
        configRepository.setLlmApiKey(provider, key)
    }

    override fun getLlmBaseUrl(provider: String): String? = configRepository.getLlmBaseUrl(provider)

    override fun setLlmBaseUrl(provider: String, url: String) {
        configRepository.setLlmBaseUrl(provider, url)
    }

    override fun getAllConfig(): AppConfig = configRepository.getConfig()
}

File: desktop/src/services/ConfigDomainService.ts
import type { AppConfig } from '../domain/configuration/entities';
import type { ConfigService } from '../domain/configuration/ports';
import type { ConfigRepository } from '../domain/configuration/ports';

export class ConfigDomainService implements ConfigService {
  constructor(private configRepository: ConfigRepository) {}

  getTargetLanguage(): string {
    return this.configRepository.getTargetLanguage();
  }

  setTargetLanguage(lang: string): void {
    this.configRepository.setTargetLanguage(lang);
  }

  getLlmProvider(): string {
    return this.configRepository.getLlmProvider();
  }

  setLlmProvider(provider: string): void {
    this.configRepository.setLlmProvider(provider);
  }

  getLlmApiKey(provider: string): string | undefined {
    return this.configRepository.getLlmApiKey(provider);
  }

  setLlmApiKey(provider: string, key: string): void {
    this.configRepository.setLlmApiKey(provider, key);
  }

  getLlmBaseUrl(provider: string): string | undefined {
    return this.configRepository.getLlmBaseUrl(provider);
  }

  setLlmBaseUrl(provider: string, url: string): void {
    this.configRepository.setLlmBaseUrl(provider, url);
  }

  getAllConfig(): AppConfig {
    return this.configRepository.getConfig();
  }
}
